### PR TITLE
Add a gRPC-backed shim implementation.

### PR DESCRIPTION
**Pull request by pgavlin**
This implementation of the shim uses the TF plugin gRPC interface to
implement provider operations. The gRPC interface exposes its own schema
system that is independent of the TF plugin SDK and communicates
configuration and state using `msgpack`-encoded `cty` values. The bulk
of the work is in translating the gRPC schema to the shim's schema,
encoding and decoding cty values to and from Go values, and computing
flatmap-style diffs.

### CODE REVIEW

**Comment by pgavlin** [comment]
Gentle ping.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
           GOPROXY: "https://proxy.golang.org"
         run: go mod download
       - name: Build
-        run: go build ./...
+        run: go build ./... && go install ./cmd/...
```
Comment (stack72) [inline .github/workflows/ci.yml:25]: why are you needing to go install here?

**Review by stack72 — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (pgavlin) [reply .github/workflows/ci.yml:25]: These changes add a provider binary that needs to be on the `PATH` when running tests; this is the easiest way to achieve that.

Comment (pgavlin) [reply .github/workflows/ci.yml:25]: Somewhat related: why does this repo no longer use the `Makefile`? IMO it's bad practice for CI and local development workflows to diverge at their core.

Comment (stack72) [reply .github/workflows/ci.yml:25]: That's a pure error - I will fix that up with our CI scripts

Comment (pgavlin) [reply .github/workflows/ci.yml:25]: Cool. Thanks!

**Pull request merged**